### PR TITLE
Implement missing calls for invoice payment resource

### PR DESCRIPTION
**Pull request by wwjfrodriguez**
Ref  [killbill-client-ruby Issue #36](https://github.com/killbill/killbill-client-ruby/issues/36)

### CODE REVIEW

File: lib/killbill_client/models/invoice_payment.rb [hunk 1]
```diff
@@ -69,6 +71,22 @@ def bulk_create(external_payment = false, user = nil, reason = nil, comment = ni
...
                         }.merge(options)
       end
+
+      def chargeback_reversal(payment_id, amount, adjustments = nil, user = nil, reason = nil, comment = nil, options = {})
+        payload             = InvoicePaymentTransactionAttributes.new
+        payload.amount      = amount
+        payload.is_adjusted = !adjustments.nil?
+        payload.adjustments = adjustments
+        invoice_payment = self.class.post "#{KILLBILL_API_INVOICE_PAYMENTS_PREFIX}/#{payment_id}/chargebackReversals",
+                        to_json,
```
Comment (pierre) [inline lib/killbill_client/models/invoice_payment.rb]: `to_json ` would need to be updated to `payload.to_json` in the `post` (it's not in use today).

File: lib/killbill_client/models/invoice_payment.rb [hunk 2]
```diff
@@ -69,6 +71,22 @@ def bulk_create(external_payment = false, user = nil, reason = nil, comment = ni
                             :comment => comment,
                         }.merge(options)
       end
+
+      def chargeback_reversal(payment_id, amount, adjustments = nil, user = nil, reason = nil, comment = nil, options = {})
```
Comment (pierre) [inline lib/killbill_client/models/invoice_payment.rb]: For symmetry with the `refund` API, I would declare this as a class method above (`class << self` block) instead.

File: lib/killbill_client/models/invoice_payment.rb [hunk 3]
```diff
@@ -69,6 +71,22 @@ def bulk_create(external_payment = false, user = nil, reason = nil, comment = ni
...
+        payload             = InvoicePaymentTransactionAttributes.new
+        payload.amount      = amount
+        payload.is_adjusted = !adjustments.nil?
```
Comment (pierre) [inline lib/killbill_client/models/invoice_payment.rb]: This is not part of the API, is it?

Only a `transactionExternalKey` and an `effectiveDate` are required (https://github.com/killbill/killbill/blob/6b092a0ca5d812ac6acab938b2b874b59c9abcf9/jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/InvoicePaymentResource.java#L257).

**Comment by wwjfrodriguez** [comment]
@pierre I think `def chargeback` is already implemented [here](https://github.com/killbill/killbill-client-ruby/blob/558844192ab85706e366aab52d2a5ba0bb43c3c9/lib/killbill_client/models/transaction.rb#L109). Excuse me if I'm wrong.

**Comment by pierre** [comment]
> @pierre I think `def chargeback` is already implemented [here](https://github.com/killbill/killbill-client-ruby/blob/558844192ab85706e366aab52d2a5ba0bb43c3c9/lib/killbill_client/models/transaction.rb#L109). Excuse me if I'm wrong.

This implements https://github.com/killbill/killbill/blob/3931c164c626fc567455c58fc27db767c179a08f/jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/PaymentResource.java#L532-L556 (create a chargeback for a payment).

I'm referring to https://github.com/killbill/killbill/blob/3931c164c626fc567455c58fc27db767c179a08f/jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/InvoicePaymentResource.java#L207-L233 (create a chargeback for an invoice payment, i.e. a payment associated with an invoice).

**Comment by wwjfrodriguez** [comment]
> This implements https://github.com/killbill/killbill/blob/3931c164c626fc567455c58fc27db767c179a08f/jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/PaymentResource.java#L532-L556 (create a chargeback for a payment).
I'm referring to https://github.com/killbill/killbill/blob/3931c164c626fc567455c58fc27db767c179a08f/jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/InvoicePaymentResource.java#L207-L233 (create a chargeback for an invoice payment, i.e. a payment associated with an invoice).

Ok @pierre, I see! Sorry for the mistake.

**Comment by wwjfrodriguez** [comment]
@pierre Yesterday I forgot to push my latest changes in this branch, sorry.

File: lib/killbill_client/models/invoice_payment.rb [hunk 4]
```diff
@@ -39,6 +41,39 @@ def refund(payment_id, amount, adjustments = nil, user = nil, reason = nil, comm
 
           invoice_payment.refresh(options)
         end
+
+        def create_chargeback(payment_id, amount, currency, effective_date = nil, user = nil, reason = nil, comment = nil, options = {})
```
Comment (pierre) [inline lib/killbill_client/models/invoice_payment.rb]: Let's name it `chargeback` for symmetry with the other methods (`refund`, `chargeback_reversal`, etc.).

Comment (pierre) [reply lib/killbill_client/models/invoice_payment.rb]: Bump - just wanted to make sure you saw this.

Comment (wwjfrodriguez) [reply lib/killbill_client/models/invoice_payment.rb]: Sorry @pierre , I did't saw it. I already pushed the branch with the change.

**Pull request merged**